lib/bowling.rb

```diff
@@ -0,0 +1,48 @@
+class Bowling
+  def initialize
+    @rolls = []
+  end
+  
+  def roll(pins)
+    raise 'You must enter a score less than 10' if pins > 10
+    @rolls << pins
+  end
+
+  def score
+    current_roll = 0
+    total = 0
+    10.times do 
+      if strike?(current_roll)
+        total += strike_bonus(current_roll)
+        current_roll += 1
+      elsif spare?(current_roll)
+        total += spare_bonus(current_roll)
+        current_roll += 2
+      else
+        total += sum_of_frame_rolls(current_roll)
+        current_roll += 2
+      end
+    end
+    total
+  end
+  
+  def spare_bonus(current_roll)
+    @rolls[current_roll + 2] + 10
+  end
+  
+  def strike_bonus(current_roll)
+    @rolls[current_roll + 1] + @rolls[current_roll + 2] + 10
+  end
+  
+  def spare?(current_roll)
+    @rolls[current_roll] + @rolls[current_roll + 1] == 10
+  end
+  
+  def strike?(current_roll)
+    @rolls[current_roll] == 10
+  end
```

Comment:
Unsure on this, but if current_roll can be either a first or second roll. Does that mean if the second roll == 10, then it's a strike? As i'm not 100% certain but a strike can only happen on the first roll of a frame. May need to look into splitting up the rolls into two different rolls per frame

Reply:
I like the way you've accounted for spares & strikes and how to accrue bonuses!